website/source/guides/security/vault-pki-integration.html.md

```diff
@@ -0,0 +1,550 @@
+configuration to a location you prefer. This environment assumes you have
+created a `templates` and `certs` directory in `/opt/nomad`.
+
+!> Note: we have hard coded the token we created into the Consul Template
+configuration file. Although we can avoid this by assigning it to the
+environment variable `VAULT_TOKEN`, this method can still pose a security
+concern. The recommended approach is to securely introduce this token to Consul
+Template. To learn how to accomplish this, see [Secure
+Introduction][secure-introduction].
+
+### Step 10: Create Templates
+
+The following are the templates used by the configuration file we specified in
+the previous step:
+
+**For Nomad Servers**:
+
+*agent.crt.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "common_name=server.global.nomad" "ttl=24h" "alt_names=localhost" "ip_sans=127.0.0.1"}}
+{{ .Data.certificate }}
+{{ end }}
+```
+
+*agent.key.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "common_name=server.global.nomad" "ttl=24h" "alt_names=localhost" "ip_sans=127.0.0.1"}}
+{{ .Data.private_key }}
+{{ end }}
+```
+
+*ca.crt.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "common_name=server.global.nomad" "ttl=24h"}}
+{{ .Data.issuing_ca }}
+{{ end }}
+```
+
+**For Nomad Clients**:
+
+Replace the word `server` in the `common_name` option in each template with the
+word `client`:
+
+*agent.crt.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "common_name=client.global.nomad" "ttl=24h" "alt_names=localhost" "ip_sans=127.0.0.1"}}
+{{ .Data.certificate }}
+{{ end }}
+```
+
+*agent.key.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "common_name=client.global.nomad" "ttl=24h" "alt_names=localhost" "ip_sans=127.0.0.1"}}
+{{ .Data.private_key }}
+{{ end }}
+```
+
+*ca.crt.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "common_name=client.global.nomad" "ttl=24h"}}
+{{ .Data.issuing_ca }}
+{{ end }}
+```
+
+**For Nomad CLI (provide this on all nodes)**:
+
+*cli.crt.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "ttl=24h" }}
+{{ .Data.certificate }}
+{{ end }}
+```
+
+*cli.key.tpl*:
+
+```
+{{ with secret "pki_int/issue/nomad-cluster" "ttl=24h" }}
+{{ .Data.private_key }}
+{{ end }}
+```
+
+### Step 10: Start the Consul Template Service
+
+Once you have written the individual templates to the `source` locations you
+specified in each of the template stanzas in [Step
+9](#step-9-configure-consul-template-on-all-nodes), you may start the Consul
+Template service on each node:
+
+```shell
+$ sudo systemctl start consul-template
+```
+You can quickly confirm the appropriate certs and private keys were generated in
+the `destination` directory you specified in your Consul Template configuration
+by listing them out:
+
+```
+$ ls /opt/nomad/certs/
+agent.crt  agent.key  ca.crt  cli.crt  cli.key
+```
+
+### Step 11: Configure Nomad to Use TLS
+
+Add the following [tls stanza][nomad-tls-stanza] to your Nomad agent
```

Comment:
Should add a line that this applies to all client nodes.

Does this not apply to the master/server nodes?

Would be good to makes sure we stay consistent in terminology, agent nodes => client nodes, master nodes => server nodes.

Reply:
My understanding is every node in the Nomad cluster is running an agent. This agent can either run in server or client mode (when you're running in dev mode, you're running the agent in both client and server mode). We don't really use the master/slave terminology.

Reply:
Thinking how I can clarify this. I could say "Add the following...to the configuration on all nodes in your cluster." How does that sound?